tests/unit/test_profiler_config.py

```diff
@@ -643,3 +646,111 @@ def test_validation():
 
     with pytest.raises(AssertionError, match=ErrorMessages.INVALID_CPROFILE_TIMER.value):
         PythonProfilingConfig(cprofile_timer="bad_cprofile_timer")
+
+
+DATA_DIR = os.path.join(os.path.dirname(__file__), "..", "data")
+SCRIPT_PATH = os.path.join(DATA_DIR, "dummy_script.py")
+INSTANCE_COUNT = 1
+INSTANCE_TYPE = "ml.p3.2xlarge"
+ROLE = "Dummy"
+REGION = "us-west-2"
+
+
+def test_create_pytorch_estimator_with_framework_profile(
+    sagemaker_session,
+    pytorch_inference_version,
+    pytorch_inference_py_version,
+    default_framework_profile,
+):
+    profiler_config = ProfilerConfig(framework_profile_params=default_framework_profile)
+
+    pytorch = PyTorch(
+        entry_point=SCRIPT_PATH,
+        framework_version=pytorch_inference_version,
+        py_version=pytorch_inference_py_version,
+        role=ROLE,
+        sagemaker_session=sagemaker_session,
+        instance_count=INSTANCE_COUNT,
+        instance_type=INSTANCE_TYPE,
+        base_job_name="job",
+        profiler_config=profiler_config,
+    )
+    assert pytorch._framework_name == "pytorch"
```

Comment:
I don't follow what this test is actually check.

Reply:
Do we have another tests that runs through as well? If you want to test warning message you can use follow mechanics:
https://docs.python.org/3/library/warnings.html#testing-warnings

Reply:
+1, can we add unit test coverage for checking the warning messages?

Reply:
Absolutely, will do.